addon/components/power-select-infinity.ts

```diff
@@ -0,0 +1,373 @@
+     * the bottom of the list is reached.
+     *
+     * @type {(keyword: string | null) => any[]}
+     * @argument loadMore
+     * @memberof PowerSelectInfinityArgs
+     */
+    loadMore?: (keyword?: string) => Promise<T[]>;
+    /**
+     * The message shown when no options are returned.
+     *
+     * @type {string}
+     * @argument noMatchesMessage
+     * @memberof PowerSelectInfinityArgs
+     */
+    noMatchesMessage?: string;
+
+    /**
+     * The delay for invoking the `@search` method
+     * when typing.
+     *
+     * @type {number}
+     * @argument searchDebounceDelay
+     * @memberof PowerSelectInfinityArgs
+     */
+    searchDebounceDelay?: number;
+    /**
+     * Used by ember-vertical-collection for occlusion rendering.
+     *
+     * @type {boolean}
+     * @argument staticHeight
+     * @memberof PowerSelectInfinityArgs
+     */
+    staticHeight?: boolean;
+}
+
+export default class PowerSelectInfinity<T> extends Component<PowerSelectInfinityArgs<T>> {
+    get allowClear() {
+        return this.args.allowClear ?? true;
+    }
+
+    get bufferSize() {
+        return this.args.bufferSize ?? 5;
+    }
+
+    get canLoadMore() {
+        return this.args.canLoadMore ?? true;
+    }
+
+    get estimateHeight() {
+        return this.args.estimateHeight ?? 30;
+    }
+
+    get loadingComponent() {
+        return this.args.loadingComponent ?? 'power-select-infinity/loading';
+    }
+
+    get matchTriggerWidth() {
+        return this.args.matchTriggerWidth ?? true;
+    }
+
+    get noMatchesMessage() {
+        return this.args.noMatchesMessage ?? 'No results found';
+    }
+
+    get optionsComponent() {
+        return this.args.optionsComponent ?? 'power-select-infinity/options';
+    }
+
+    get search() {
+        return this.args.search ?? false;
+    }
+
+    get searchDebounceDelay() {
+        return this.args.searchDebounceDelay ?? 300;
+    }
+
+    get searchEnabled() {
+        return this.args.searchEnabled ?? true;
+    }
+
+    get searchField() {
+        return this.args.searchField ?? undefined;
+    }
+
+    get staticHeight() {
+        return this.args.staticHeight ?? false;
+    }
+
+    get tabindex() {
+        return this.args.tabindex ?? -1;
+    }
+
+    @tracked isLoadingMore = false;
+    @tracked loading = false;
+    @tracked searchText = '';
+    @tracked select: Select | null = null;
+
+    get showCreateMessage() {
+        const searchText = this.select?.searchText;
+        const numOptions = guard(this.args.options, 'content')
+            ? this.args.options.content.length
+            : this.args.options.length;
+        return this.args.canCreate && numOptions === 0 && searchText !== '' && !this.task.isRunning;
+    }
+
+    get triggerComponent() {
+        return this.args.triggerComponent ?? 'power-select-infinity/trigger-search';
+    }
+
+    get triggerClass() {
+        return 'ember-power-select-infinity-trigger';
```

Comment:
Probably want to allow this to have an argument passed in as well. Like you did for the `get dropdownClass()` below.